components/registry-facade/pkg/registry/registry.go

```diff
@@ -253,51 +255,47 @@ func NewRegistry(cfg config.Config, newResolver ResolverProvider, reg prometheus
 }
 
 func getRedisClient(cfg *config.RedisCacheConfig) (*redis.Client, error) {
-	if cfg.SingleHostAddress != "" {
-		log.WithField("addr", cfg.SingleHostAddress).WithField("username", cfg.Username).Info("connecting to single Redis host")
-		rdc := redis.NewClient(&redis.Options{
-			Addr:     cfg.SingleHostAddress,
-			Username: cfg.Username,
-			Password: cfg.Password,
-		})
-
-		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
-		defer cancel()
-
-		_, err := rdc.Ping(ctx).Result()
-		if err != nil {
-			return nil, xerrors.Errorf("cannot check Redis connection: %w", err)
-		}
+	if cfg.SingleHostAddress == "" {
+		return nil, xerrors.Errorf("registry-facade setting 'singleHostAddr' is missing")
+	}
 
-		return rdc, nil
+	opts := &redis.Options{
+		Addr: cfg.SingleHostAddress,
 	}
 
-	if cfg.MasterName == "" {
-		return nil, fmt.Errorf("redis masterName must not be empty")
+	if cfg.Username != "" {
+		opts.Username = cfg.Username
 	}
-	if len(cfg.SentinelAddrs) == 0 {
-		return nil, fmt.Errorf("redis sentinelAddrs must not be empty")
+
+	if cfg.Password != "" {
+		opts.Password = cfg.Password
 	}
 
-	rdc := redis.NewFailoverClient(&redis.FailoverOptions{
-		MasterName:    cfg.MasterName,
-		SentinelAddrs: cfg.SentinelAddrs,
-		Username:      cfg.Username,
-		Password:      cfg.Password,
+	if cfg.UseTLS && cfg.TLSCertificate != "" {
+		certs := x509.NewCertPool()
+		certs.AppendCertsFromPEM([]byte(cfg.TLSCertificate))
 
-		SentinelUsername: cfg.Username,
-		SentinelPassword: cfg.Password,
+		tlsConfig := &tls.Config{
+			RootCAs:    certs,
+			MinVersion: tls.VersionTLS12,
```

Comment:
I am not sure if we should enforce this strong criterion.
To me, I think verifying client CA and root CA is enough.